### ISSUE TITLE
yabridge crashes with MT Power Drum Kit (WineD3D issue) [with workaround]

### ISSUE THREAD

**Issue by meddario** [robbert-vdh/yabridge#326]
### Thanks for giving yabridge a shot!

- [X] I read through both the troubleshooting and the known issues sections, and my issue wasn't listed there

## :information_source:  Temporary solution

Since the problem is related to graphics drivers, one solution is to use software rendering, MT Power Drum Kit also provides "**MT-PowerDrumKit_Alternative.vst3**", whose GUI works correctly. 

### Problem description

Using Reaper 7.12, adding the [MT Power Drum Kit VST3 plugin](https://www.powerdrumkit.com/download76187.php) on a track results in a yabridge crash.

```
yabridgectl --version 
yabridgectl 5.1.0

wine --version
wine-9.4 (Staging)
```

### What did you expect to happen?

I exptected the VST to be correctly added to the track

### What actually happened?

Yabridge crashed and then Reaper crashed

### Operating system

Ubuntu 22.04.4 LTS

### How did you install yabridge?

From release https://github.com/robbert-vdh/yabridge/releases/download/5.1.0/yabridge-5.1.0.tar.gz

### yabridge version

5.1.0

### yabridgectl version

5.1.0

### Wine version

wine-9.4 (Staging)

### Plugin

MT Power Drum Kit

### Plugin type

VST3

### Plugin architecture

64-bit

### Host

Reaper 7.12

### Desktop environment or WM

KDE Plasma

### Debug log

[yabridge.log](https://github.com/robbert-vdh/yabridge/files/14817786/yabridge.log)
[backtrace.txt](https://github.com/robbert-vdh/yabridge/files/14817787/backtrace.txt)

**Comment by jbmestelan** [comment]
Same here on Fedora 39 + Ardour. 

winehq-staging-1:9.4-1.1
yabridge-5.1.0-6.fc39

This setup was recently working; upgraded to yabridge 5.1.0-6 happened this month: this may be the cause of the error (unsure whether I had tried before).

**Comment by meddario** [comment]
In my case the plugin was working (same plugin version and yabridge version), I also noticed a strange thing: the vst still works in an existing project, it's just seemingly impossible to add it to a new track. I just got a wine-staging update, I'll try again.

**Comment by robbert-vdh** [comment]
@meddario There is nothing indicating an issue in the yabridge log, and the Wine backtrace points to WineD3D. If this started happening recently after a system update, there may just be something wrong with your OpenGL drivers. Make sure you can run `glxgears`, and check that `glxinfo` lists your GPU (i.e. it's not falling back to software rendering).

For what it's worth, both the 64-bit VST2 and the 64-bit VST3 versions of MT-PowerDrumKit 2.1.2 work fine for me with REAPER 7.12, wine-tkg 9.4, and the latest master branch version of yabridge.

**Comment by meddario** [comment]
@robbert-vdh thanks for the feedback, I'll investigate those things. For the record, I downgraded Wine to version 9.3 with:

```sh
#! /usr/bin/env bash

set -e

WINE_DOWNGRADE_VERSION=9.3~jammy-1

echo "Downgrading Wine Staging to version $WINE_DOWNGRADE_VERSION"

apt update
apt install -y wine-staging=$WINE_DOWNGRADE_VERSION \
    wine-staging-amd64=$WINE_DOWNGRADE_VERSION \
    wine-staging-i386=$WINE_DOWNGRADE_VERSION \
    winehq-staging=$WINE_DOWNGRADE_VERSION
apt-mark hold winehq-staging
```
and the problem is still present, so probably it's not related to yabridge.

I'll update here if i find out something interesting.

**Comment by meddario** [comment]
@robbert-vdh I didn't find other useful information, do you suggest to report this issue directly to Wine?

**Comment by robbert-vdh** [comment]
Did this plugin work before? You can try figuring out what you updated that caused this to break. Or you can try the 'alternative' version of the plugin which IIRC doesn't use hardware accelerated graphics (though that of course won't solve the underlying issue).

**Comment by meddario** [comment]
@robbert-vdh yes, the plugin was working before (GUI included), and the sound still is (in existing projects), it breaks when I try to interact with the GUI.
In the meanwhile, I tried to use the "MT-PowerDrumKit_Alternative.vst3", and with that the GUI works without crashes. 
At this point the problem is for sure with graphics drivers, or even with the hardware itself. I already had problems with this laptop's GPU under Windows, where it's basically unusable and overheats, on Ubuntu the situation is (was) better and I was able to complete a few recordings, but maybe it has just reached its end of life... Anyway, thank you very much for the help (and for this great project), I'll post here if I discover something useful in the next days.
